backend/src/main/java/ai/giskard/service/TestSuiteService.java

```diff
@@ -49,6 +49,19 @@ public class TestSuiteService {
     private final MLWorkerWSCommService mlWorkerWSCommService;
     private final TestFunctionRepository testFunctionRepository;
 
+    public Long saveTestSuite(String projectKey, TestSuiteDTO dto) {
+        if (dto.getName() == null || dto.getName().isEmpty()) {
+            throw new IllegalArgumentException("Test suite name cannot be empty");
```

Comment:
Nit pick, you can use `Strings.isEmpty()` util that is `null` safe. 

IMO we should check for blank to avoid name like `  `

```suggestion
        if (Strings.isBlank()) {
            throw new IllegalArgumentException("Test suite name cannot be blank");
```